engine/execution/ingestion/engine.go

```diff
@@ -423,6 +423,32 @@ func (e *Engine) BlockProcessable(b *flow.Header) {
 	}
 }
 
+// BlockFinalized handles the finalized blocks
+func (e *Engine) BlockFinalized(b *flow.Header) {
+	blockID := b.ID()
+	newBlock, err := e.blocks.ByID(blockID)
```

Comment:
are we able to query seals only, rather than the entire block? Since we only need the seals here

Reply:
I tried that initially but getting seals by blockId was broken and returns only one seal which I think it has a different meaning.

Reply:
I assume you are referring to this function: https://github.com/onflow/flow-go/blob/0f54772a1af3dee646c2da58b7edd0221ef70d35/storage/seals.go#L19-L20

I don't think it is broken. But it probably has a different meaning than what you are expecting: 
* `ByBlockID(B)` it does _not_ mean: give me the seal for block `B`.
* it returns you the latest seal that was included in the fork with head `B`. So for example, if you take the following fork:
    - ` <- B0 <- B1 <- B2 <- B3{seals B0} <- B4 <- B`
    - `ByBlockID(B)` will return the seal for B0, because `B0` is the latest sealed block in the fork with head `B`

Reply:
Right, so we don't have a way to query all seals in the payload of a block. I think what @ramtinms has implemented here is correct.